parse_nucleotide_fasta.c

```diff
@@ -0,0 +1,263 @@
+                printf("\t-file <file> : Directs the program to the fasta file.\n");
+                printf("\t-gc : global gc count.\n");
+                printf("\t-gc=<x> : gc count per x bases.\n");
+                printf("\t-k_mers <k> : lists all k-mers of size k.\n");
+                printf("\t-match <string> : searches for all instances of a string. (only words for same case searches)\n");
+                argument_legality = -1;
+                break;
+            case '?':
+                printf("Unknown or incorrect argument entered, exiting program. Please see maunal or enter ./parse_nuc_fasta -help\n");
+                argument_legality = -1;
+                break;
+            default:
+                printf("No legal options entered. Please see maunal or enter ./parse_nuc_fasta -help\n");
+                argument_legality = -1;
+                break;
+        }
+    }
+    if (flags[0].flag_raised == 0) {
+        printf("A fasta file must be passed in. Please see maunal or enter ./parse_nuc_fasta -help\n");
+        argument_legality = -1;
+    } else if (flags[1].flag_raised == 0 &&
+               flags[2].flag_raised == 0 &&
+               flags[3].flag_raised == 0) {
+        printf("No process arguments entered, select a process for the program carry out. Please see maunal or enter ./parse_nuc_fasta -help\n");
+        argument_legality = -1;
+    }
+    return argument_legality;
+}
+
+/******************************
+*   Counts the GC content in the read passed into the function. 
+*******************************/
+int GC_count(struct Flags* flags, char* read, int* site_values, int chunk_num, int chunk_val) {
+    // counts GC content per X bases
+    int site_index = 0;
+    while (read[site_index] != '\0') {
+        if(site_values[4] == chunk_val && chunk_val != 0) {
+            printf("Chunk number: %d A: %d T: %d C: %d G:%d\n", chunk_num, site_values[0], site_values[1], site_values[2], site_values[3]);
+            printf("Chunk: %d GC content: %f\n", chunk_num, (((float)site_values[2] + (float)site_values[3]) / (float)site_values[4]) * 100);
+            int i;
+            for (i = 0; i < 5; i++) {
+                site_values[i] = 0;
+            }
+            chunk_num++;
+        }
+        char site = read[site_index];
+        if (site == 'A' || site == 'a') {
+            site_values[0]++;
+            site_values[4]++;
+        } else if (site == 'T' || site == 't') {
+            site_values[1]++;
+            site_values[4]++;
+        } else if (site == 'C' || site == 'c') {
+            site_values[2]++;
+            site_values[4]++;
+        } else if (site == 'G' || site == 'g') {
+            site_values[3]++;
+            site_values[4]++;
+        } else {
+          // Do Nothing
+        }
+        site_index++;
+    }
+
+    return chunk_num;
+}
+
+int find_all_kmer_permutations() {
+    // Finds all the permutations for all K-mers of size K
```

Comment:
Hrmmmmmm, did you mean to leave this in here?

Reply:
It doesn't really matter, but usually you wouldn't merge incomplete code into master. You'd work on it out on a branch and then when it's finished, merge it in.